### PR TITLE
Validate put and post routes

### PR DESCRIPTION
**Pull request by Starkandco**
Joi validation on put and post requests for user and login routes

### CODE REVIEW

**Comment by Starkandco** [comment]
@grizzle2101 - I didn't know should I do the same for the get routes as the format of these is slightly different? Sure let me know if there's anything you want to change anyway

File: backend/src/routes/login.route.ts [hunk 1]
```diff
@@ -15,8 +16,12 @@ export class LoginRoute {
 
   private initializeRoutes() {
     this.router.post(this.path, async (req, res) => {
+
+      const requestValidity = this.validateRequest(req.body);
+      if (requestValidity.error) return res.status(404).send(requestValidity.error.message);
+        
       const user = await this.users.findOne({
-        email: req.body.email,
+        email: req.body.email.toLowerCase(),
```
Comment (grizzle2101) [inline backend/src/routes/login.route.ts:24]: Do you want to put this into the JOI validation?

**Review by grizzle2101 — CHANGES_REQUESTED** [review]
one small thing there with the to. Lower(), id plop that into the validator instead.

File: backend/src/routes/login.route.ts (hunk 1 above)
Comment (Starkandco) [reply backend/src/routes/login.route.ts]: Got it working there

**Review by grizzle2101 — APPROVED** [review]

**Pull request merged**